test/e2e/namespace_test.go

```diff
@@ -139,4 +139,107 @@ var _ = ginkgo.Describe("[namespace auto-provision] namespace auto-provision tes
 			})
 		})
 	})
+
+	ginkgo.When("unjoining not ready cluster", func() {
+		clusterName := "member-e2e-" + rand.String(3)
+		homeDir := os.Getenv("HOME")
+		kubeConfigPath := fmt.Sprintf("%s/.kube/%s.config", homeDir, clusterName)
+		controlPlane := fmt.Sprintf("%s-control-plane", clusterName)
+		clusterContext := fmt.Sprintf("kind-%s", clusterName)
+		deploymentName := deploymentNamePrefix + rand.String(RandomStrLength)
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Creating cluster: %s", clusterName), func() {
+				err := createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Joinning cluster: %s", clusterName), func() {
+				karmadaConfig := karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+				opts := karmadactl.CommandJoinOption{
+					GlobalCommandOptions: options.GlobalCommandOptions{
+						DryRun: false,
+					},
+					ClusterNamespace:  "karmada-cluster",
+					ClusterName:       clusterName,
+					ClusterContext:    clusterContext,
+					ClusterKubeConfig: kubeConfigPath,
+				}
+				err := karmadactl.RunJoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Creating Deployment workload in cluster: %s", clusterName), func() {
+
+				deployment := helper.NewDeployment(testNamespace, deploymentName)
+				framework.CreateDeployment(kubeClient, deployment)
+			})
+		})
+
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Change cluster %s status is unready", clusterName), func() {
+				cluster, err := karmadaClient.ClusterV1alpha1().Clusters().Get(context.TODO(), clusterName, metav1.GetOptions{})
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				cluster.Status.Conditions = []metav1.Condition{
+					{
+						Status: "False",
+					},
+				}
+				_, err = karmadaClient.ClusterV1alpha1().Clusters().Update(context.TODO(), cluster, metav1.UpdateOptions{})
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
```

Comment:
Only update the conditions maybe can not cover the case, we can call `disableCluster` to disable cluster:

https://github.com/karmada-io/karmada/blob/697e120922db5b67212308ba9a3a39810a896096/test/e2e/failover_test.go#L142-L143